### PR TITLE
[next] Form Standardization

### PR DESCRIPTION
**Pull request by wyattjoh**
## What does this PR do?

Standardizes the form interfaces and usage across the application.

### CODE REVIEW

File: src/core/client/admin/test/configure/moderation.spec.tsx [hunk 1]
```diff
@@ -203,14 +203,6 @@ it("change akismet settings", async () => {
   // Input valid api key
   keyField.props.onChange("my api key");
 
-  // Input malformed site.
-  siteField.props.onChange("malformed url");
-
-  expect(
-    within(akismetContainer).queryAllByText("This field is required.").length
-  ).toBe(0);
-  expect(within(akismetContainer).queryAllByText("Invalid URL").length).toBe(1);
```
Comment (nick-funk) [inline src/core/client/admin/test/configure/moderation.spec.tsx:212]: If none of the form's validation logic is changed, why were these expectations removed?

File: src/core/client/stream/tabs/Comments/Comment/EditCommentForm/EditCommentForm.tsx [hunk 2]
```diff
@@ -70,6 +70,7 @@ const EditCommentForm: FunctionComponent<EditCommentFormProps> = props => {
               name="body"
               validate={getCommentBodyValidators(props.min, props.max)}
             >
+              {/* FIXME: (wyattjoh) reorganize this */}
```
Comment (nick-funk) [inline src/core/client/stream/tabs/Comments/Comment/EditCommentForm/EditCommentForm.tsx:73]: Was this left in as a personal reminder, or as a "defer util later" notice?

File: src/core/client/stream/tabs/Comments/Comment/ReplyCommentForm/ReplyCommentForm.tsx [hunk 3]
```diff
@@ -57,6 +57,7 @@ const ReplyCommentForm: FunctionComponent<ReplyCommentFormProps> = props => {
               name="body"
               validate={getCommentBodyValidators(props.min, props.max)}
             >
+              {/* FIXME: (wyattjoh) reorganize this */}
```
Comment (nick-funk) [inline src/core/client/stream/tabs/Comments/Comment/ReplyCommentForm/ReplyCommentForm.tsx:62]: Was this left in as a personal reminder, or as a "defer util later" notice?

**Review by nick-funk — CHANGES_REQUESTED** [review]

File: src/core/client/admin/test/configure/moderation.spec.tsx (hunk 1 above)
Comment (wyattjoh) [reply src/core/client/admin/test/configure/moderation.spec.tsx:212]: Validation logic was changed:

https://github.com/coralproject/talk/pull/2425/files#diff-194a472865e421b58154dea9572ee27dL92

File: src/core/client/stream/tabs/Comments/Comment/EditCommentForm/EditCommentForm.tsx (hunk 2 above)
Comment (wyattjoh) [reply src/core/client/stream/tabs/Comments/Comment/EditCommentForm/EditCommentForm.tsx:73]: `CORL-492`

File: src/core/client/stream/tabs/Comments/Comment/ReplyCommentForm/ReplyCommentForm.tsx (hunk 3 above)
Comment (wyattjoh) [reply src/core/client/stream/tabs/Comments/Comment/ReplyCommentForm/ReplyCommentForm.tsx:62]: `CORL-492`

**Review by nick-funk — APPROVED** [review]

**Pull request merged**